components/discount/registerDiscount.tsx

```diff
@@ -370,6 +425,17 @@ const RegisterDiscount: FunctionComponent<RegisterDiscountProps> = ({
               isSwissResident={isSwissResident}
               onSwissResidentChange={() => setIsSwissResident(!isSwissResident)}
             />
+            {couponCode ? (
```

Comment:
Yes so this is the kind of stuff we want to avoid in a component in general to make it as general as possible